tuned/profiles/functions/function_intel_recommended_pstate.py

```diff
@@ -0,0 +1,44 @@
+import os
+import tuned.logs
+from . import base
+from tuned.profiles.exceptions import InvalidProfileException
+
+log = tuned.logs.get()
+processor_name = ("sandybridge",
+        "ivybridge",
+        "haswell",
+        "broadwell",
+        "skylake")
+
+pmu_path = "/sys/devices/cpu/caps/pmu_name"
+active = "active"
+disable = "disable"
+
+class intel_recommended_pstate(base.Function):
+    """
+    Checks the processor code name and return the pstate
+    value. Active is returned for the newer generation of 
+    processors not in the processor_name list.
+    Only accepted arguments are disable and active. If any 
+    other string passed, it throws InvalidProfileException. This
+	exception will abort profile loading.
+    """
+    def __init__(self):
+        # 2 arguments
+        super(intel_recommended_pstate, self).__init__("intel_recommended_pstate", 2)
+
+    def execute(self, args):
+        if not super(intel_recommended_pstate, self).execute(args):
+            return None
+        # when no arguements are passed, we return disable and not fail    
+        if len(args) == 0:
```

Comment:
Why use arguments at all when only constants are supported? Can't the function be argument-less and always return disable/active?

Reply:
Wanted to have some error handling capabilities. The updated commit will not require any arguments to return the constant, i.e. disable|active